shell/server/proxy.js

```diff
@@ -1604,7 +1598,7 @@ Proxy.prototype.handleRequest = function (request, data, response, retryCount) {
     var path = request.url.slice(1);  // remove leading '/'
     var session = self.getSession(request);
 
-    if (request.method === "GET") {
+    if (request.method === "GET" || request.method === "HEAD") {
```

Comment:
The distinction is valuable when sending the body is an expensive operation, like when a HEAD would retrieve the size of a large file, but the GET would fetch the whole 4GB contents.  It'd be nice to not make the grain attempt to stream all that back to the proxy for no particular reason. :)